EssayMonkeySolution.py

```diff
@@ -0,0 +1,116 @@
+import os
+import random
+import sys
+__author__ = 'joanhong'
+
+
+# generate essay based on input number of paragraphs and sentences
+def generate_essay(num_parag, num_sent):
+
+    adjs = parse_text_files("EssayMonkeyAdjectives.txt")
+    nouns = parse_text_files("EssayMonkeyNouns.txt")
+    verbs = parse_text_files("EssayMonkeyVerbs.txt")
+
+    num_adjs = len(adjs)
+    num_nouns = len(nouns)
+    num_verbs = len(verbs)
+
+    # determine how many adjectives per noun
+    target = num_adjs/num_parag/num_sent
+    # keep it to a reasonable range
+    if target == 0:
+        target = 1
+    elif target > 5:
+        target = 5
+
+    punctuation = ['.', '!', '?']
+
+    for np in range(num_parag):
+        for ns in range(num_sent):
+            # track array location
+            adjs_loc = random.randrange(num_adjs)
+            nouns_loc = random.randrange(num_nouns)
+            verbs_loc = random.randrange(num_verbs)
+
+            # randomize number of adjectives for each noun
+            adj_type1 = random.randrange(target+1)
+            adj_type2 = random.randrange(target+1)
+
+            # construct the sentence
+            sentence = ""
+
+            # randomize clause type: noun-verb, verb-noun, noun-verb-noun
+            clause_type = random.randrange(2)
+            if clause_type == 0:
+                order = random.randrange(2)
+                # adj-noun-verb
+                sentence += generate_adjectives(adj_type1, adjs_loc, adjs)
+                sentence += nouns[nouns_loc] + " " + verbs[verbs_loc]
+
+                # adj-noun-verb-adj-noun
+                if order == 0:
+                    adjs_loc = random.randrange(num_adjs)
+                    sentence += " " + generate_adjectives(adj_type2, adjs_loc, adjs)
+                    nouns_loc = random.randrange(num_nouns)
+                    sentence += nouns[nouns_loc]
+            else:
+                # verb-adj-noun
+                sentence += verbs[verbs_loc] + " "
+                sentence += generate_adjectives(adj_type1, adjs_loc, adjs)
+                sentence += nouns[nouns_loc]
+            sys.stdout.write(str.capitalize(sentence) + punctuation[random.randrange(3)] + " ")
+        sys.stdout.write("\n\n")
```

Comment:
Hi Charles,

Thank you for pointing that out! I would be happy to modify the code
immediately to fix this problem. Should I do so and submit a new pull
request?

Best,
Joan Hong

On Tue, Nov 3, 2015 at 6:57 AM, Charles Christolini <
notifications@github.com> wrote:

> In EssayMonkeySolution.py
> <https://github.com/binarypie/problems/pull/7#discussion_r43756613>:
>
> > +                sentence += generate_adjectives(adj_type1, adjs_loc, adjs)
> > +                sentence += nouns[nouns_loc] + " " + verbs[verbs_loc]
> > +
> > +                # adj-noun-verb-adj-noun
> > +                if order == 0:
> > +                    adjs_loc = random.randrange(num_adjs)
> > +                    sentence += " " + generate_adjectives(adj_type2, adjs_loc, adjs)
> > +                    nouns_loc = random.randrange(num_nouns)
> > +                    sentence += nouns[nouns_loc]
> > +            else:
> > +                # verb-adj-noun
> > +                sentence += verbs[verbs_loc] + " "
> > +                sentence += generate_adjectives(adj_type1, adjs_loc, adjs)
> > +                sentence += nouns[nouns_loc]
> > +            sys.stdout.write(str.capitalize(sentence) + punctuation[random.randrange(3)] + " ")
> > +        sys.stdout.write("\n\n")
>
> Your output ends up with extra newlines at the end.
>
> > python joanhong.py
> Number of paragraphs: 5
> Number of sentences per paragraph: 5
> Come human idea! Hard parent called. Play guy! Federal woman start? Put thing.
>
> Open red available president buy? Change spent wrong past foreign fine teacher? Won international full easy clear door. Old law took? Significant similar hot room opened significant similar hot central week!
>
> Close significant similar dead life! Local others work. Different long important life called! Small national different black back seem! Small national woman go.
>
> Close moment. Turned little important political person! Poor significant similar hot law waited month! Cut friend. Nice huge girl do difficult family?
>
> Man allow? Said back. Hour should? Guy must. Clear recent name happened cultural information!
>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/binarypie/problems/pull/7/files#r43756613>.
>



-- 
*Joan Hong*
B.S. Computer Engineering & Computer Science
University of Southern California | 2018

SWE-USC | Corporate Affairs Committee
joanhong@usc.edu | (925) 789-7811